### PR TITLE
Upgrade version of React used by DevTools (15.4 -> 16.4)

### PR DESCRIPTION
**Pull request by bvaughn**
* Upgrade `react` and `react-dom` dependencies (15.4 -> 16.4)
* Replace `React.createClass` with class components
* Replace `React.PropTypes` with `prop-types` package

### CODE REVIEW

File: frontend/Node.js [hunk 1]
```diff
@@ -10,6 +10,8 @@
  */
 'use strict';
 
+const PropTypes = require('prop-types');
```
Comment (gaearon) [inline frontend/Node.js]: const/var is inconsistent but I don't care

Comment (bvaughn) [reply frontend/Node.js]: Yeah. This is the codemod's doing. It tries to match what's in the file (based on the presence of other `const`/`let` keywords). We're inconsistent in a lot of places in the DevTools codebase, unfortunately.

File: frontend/PropVal.js [hunk 2]
```diff
@@ -149,49 +149,47 @@ function previewProp(val: any, nested: boolean, inverted: boolean, theme: Theme)
 }
 
 function previewArray(val, inverted, theme) {
-  var items = {};
+  var items = [];
   val.slice(0, 3).forEach((item, i) => {
-    items['n' + i] = <PropVal val={item} nested={true} inverted={inverted} theme={theme} />;
-    items['c' + i] = ', ';
+    if (i > 0) {
+      items.push(<span key={i+','}>{', '}</span>);
```
Comment (gaearon) [inline frontend/PropVal.js]: Do we ever need a span here? Seems like just text would do.

Comment (bvaughn) [reply frontend/PropVal.js]: Huh, at first I thought we did– to preserve white space– but I think you're right 😄

**Pull request merged**